src/components/NavBar/NavBar.tsx

```diff
@@ -19,25 +20,43 @@ const useStyles = makeStyles<Theme>(theme => ({
 
 const NavBar: React.FC<NavBarProps> = props => {
   const classes = useStyles();
+  const [isMenuVisible, updateMenuVisibility] = useState(false);
 
   const button: NavBarButton = {
     menu: () => <MenuIcon />,
     back: () => <ArrowBackIcon />,
   };
   const { type, onClick } = props;
 
+  const showSlideMenu = () => {
+    updateMenuVisibility(!isMenuVisible);
+  };
+
+  const handleOnClick = () => {
+    onClick ? onClick() : showSlideMenu();
+  };
+
+  const handleOnClose = () => {
+    updateMenuVisibility(false);
+  };
+
   return (
-    <AppBar className={classes.AppBar} color={'inherit'} position='fixed'>
-      <Toolbar>
-        <IconButton
-          edge='start'
-          className={classes.menuButton}
-          color='inherit'
-          onClick={onClick}>
-          {button[type]()}
-        </IconButton>
-      </Toolbar>
-    </AppBar>
+    <>
+      <AppBar className={classes.AppBar} color={'inherit'} position='fixed'>
+        <Toolbar>
+          <IconButton
+            edge='start'
+            className={classes.menuButton}
+            color='inherit'
+            onClick={handleOnClick}>
+            {button[type]()}
+          </IconButton>
+        </Toolbar>
+      </AppBar>
+      {!onClick && (
```

Comment:
The drawer menu shouldn't be removed, it's always present in the app after login.
What's the purpose of `!onClick &&` condition?

Reply:
Well I was sure, that menu is only triggered from hamburger icon - check CoS in https://digitalrig.atlassian.net/jira/software/projects/BUD/boards/497?selectedIssue=BUD-34. That's why it supports `menu` and `back` types. It's reflected in Figma and that's why implementation looks like.  But yeah, menu is only available when `<Navbar type='menu'>`.  

We can think about improvements and it available all the time also with support for swiping as separate task. I think it's better to focus more on core features development than those _nice to have_ ones. Let's create a task so we remember to do that. What do you think?

Reply:
Sure, just add a new task to the backlog